### PR TITLE
Apache decoder fix

### PR DESCRIPTION
**Pull request by e5l**

### CODE REVIEW

File: ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/DownloadTest.kt [hunk 1]
```diff
@@ -34,8 +33,20 @@ class DownloadTest : ClientLoader() {
...
             assertTrue(size > 0)
         }
     }
+
+    @Test
+    fun testDownload8175() = clientTests {
+        test { client ->
+            repeat(100) {
+                val uri =
+                    "https://raw.githubusercontent.com/ssp/ktor/4f1986df68e3594714ea12949c8af8274be99d01/ktor-client/ktor-client-apache/jvm/test/io/ktor/client/engine/apache/8175.txt"
```
Comment (dtretyakov) [inline ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/DownloadTest.kt:47]: We should store this test file in official ktor/ktor repository.

File: ktor-io/jvm/src/io/ktor/utils/io/ByteBufferChannel.kt [hunk 2]
```diff
@@ -1552,9 +1558,11 @@ internal open class ByteBufferChannel(
         }
     }
 
-    override suspend fun write(min: Int, block: (ByteBuffer) -> Unit) {
+    override fun writeAvailable(min: Int, block: (ByteBuffer) -> Unit): Int {
         require(min > 0) { "min should be positive" }
+        require(min <= 4088) { "min should be positive" }
```
Comment (dtretyakov) [inline ktor-io/jvm/src/io/ktor/utils/io/ByteBufferChannel.kt:1563]: The error message should reflect the number `4088`.

Comment (dtretyakov) [reply ktor-io/jvm/src/io/ktor/utils/io/ByteBufferChannel.kt:1563]: Maybe we need to introduce a constant for this number?

File: ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/DownloadTest.kt (hunk 1 above)
Comment (e5l) [reply ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/DownloadTest.kt:47]: Sure. I will replace it with the generated data of specific length.

File: ktor-io/jvm/src/io/ktor/utils/io/ByteBufferChannel.kt (hunk 2 above)
Comment (e5l) [reply ktor-io/jvm/src/io/ktor/utils/io/ByteBufferChannel.kt:1563]: Sure, will do

**Pull request merged**